src/compiler/checker.ts

```diff
@@ -13514,8 +13514,11 @@ namespace ts {
                         // Two conditional types 'T1 extends U1 ? X1 : Y1' and 'T2 extends U2 ? X2 : Y2' are related if
                         // one of T1 and T2 is related to the other, U1 and U2 are identical types, X1 is related to X2,
                         // and Y1 is related to Y2.
-                        if (isTypeIdenticalTo((<ConditionalType>source).extendsType, (<ConditionalType>target).extendsType) &&
-                            (isRelatedTo((<ConditionalType>source).checkType, (<ConditionalType>target).checkType) || isRelatedTo((<ConditionalType>target).checkType, (<ConditionalType>source).checkType))) {
+                        const u1 = instantiateType((<ConditionalType>source).extendsType, reportUnmeasurableMarkers);
+                        const u2 = instantiateType((<ConditionalType>target).extendsType, reportUnmeasurableMarkers);

```

Comment:
The source and target are identical sans type parameter instantiations during variance checking, so I've been wondering if we can get away with only instantiating one of the source or target.

Reply:
Yeah that feels like it should work. At least, I don't see how replacing a type parameter with a sub/super type of itself could affect whether it gets instantiated.